connect/clients/ipfs.py

```diff
@@ -0,0 +1,124 @@
+            with open('payload.json', 'w') as payload_temp_file:
+                json.dump(payload_dict, payload_temp_file)
+
+            files = {'upload-file': ('payload.json', open('payload.json', 'r'),
+                     'multipart/form-data')}
+            response = await client.post(self._add_to_ipfs_cluster_uri,
+                                         files=files)
+
+            if response.status_code == 200:
+                response_dict = json.loads(response.json())
+                info_msg = "Successfully added and pinned to IPFS -- " + \
+                           f"cid: {response_dict['cid']}"
+                logger.info(info_msg)
+                return response_dict['cid']
+            else:
+                error_msg = "Error persiting to IPFS;" + \
+                            f"HTTP response_code: {response.status_code}"
+                logger.error(error_msg)
+                return None
+
+    async def get_ipfs_cluster_peers(self):
+        """
+        Returns a list of all connected IPFS Cluster Peers.
+        Note: IPFS Node Peers are managed implicitly by IPFS Cluster.
+
+        :returns: JSON with IPFS Cluster and managed Nodes information.
+                  Returns None if a non 200 response code is returned from
+                  IPFS Cluster
+        """
+        async with httpx.AsyncClient() as client:
+            response = client.get(self._get_ipfs_cluster_peers_uri)
+
+            if response.status_code is 200:
+                return response.json()
+            else:
+                error_msg = f"Error on retrieving IPFS Cluster peers;" + \
+                            f"HTTP response_code: {response.status_code}"
+                logger.error(error_msg)
+                return None
+
+    async def unpin_content_from_cluster(self, content_cid):
+        """
+        Unpins content from the IPFS cluster
+        :param content_cid
+
+        :returns int: Returns an integer code; 0 for success; 1 for failure
+        """
+        if not content_cid:
+            error_msg = "content_cid parameter not provided for unpinning"
+            logger.error(error_msg)
+            raise ValueError(error_msg)
+
+        _unpin_uri = self._unpin_cid_ipfs_cluster_uri + content_cid
+
+        async with httpx.AsyncClient() as client:
+            response = client.delete(_unpin_uri)
+            if response.status_code == 200:
+                info_msg = f"Unpinned cid: {content_cid} from IPFS Cluster"
+                logger.info(info_msg)
+                return 0
+            else:
+                error_msg = "Error unpinning content from IPFS Cluster;" + \
+                            f"HTTP response_code: {response.status_code}"
+                logger.error(error_msg)
+                return 1
+
+
+def get_ipfs_cluster_client() -> IPFSClient:
+    global _ipfs_client
+
+    if not _ipfs_client:
+        settings = get_settings()
+        _ipfs_cluster_uri = settings.ipfs_cluster_uri
```

Comment:
from a reading standpoint we may prefer to 
```
settings = get_settings()
_ipfs_client = IPFSClient(settings.ipfs_cluster_uri, 
                          settings.ipfs_cluster_replication_factor)
```

Reply:
Done